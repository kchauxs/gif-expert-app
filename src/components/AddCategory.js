import React, { useState } from 'react'
import PropTypes from 'prop-types'


export const AddCategory = ({ setCategoties }) => {

    const [inputValue, setInputValue] = useState("")

    const handleInputChange = (e) => setInputValue(e.target.value)

    const handleSumit = (e) => {
        e.preventDefault()

        if (inputValue.trim().length > 2 ) {
            setCategoties(categories => [ inputValue,...categories])
        }
        setInputValue("")
    }

    return (
        <form onSubmit={handleSumit}>
            <input type="text"
                value={inputValue}
                onChange={handleInputChange}
                placeholder="Add Category" />
        </form>
    )
}

AddCategory.protoType = {
    setCategoties:PropTypes.func.isRequired         
}
