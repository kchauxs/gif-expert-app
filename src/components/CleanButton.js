import React, { Fragment } from 'react'
import PropTypes from 'prop-types'


export const CleanButton = ({ setCategoties }) => {

    const handleClear = () => setCategoties([])

    return (
        <Fragment>
            <button onClick={handleClear} className="btn-clean">Clear </button>
        </Fragment>
    )
}

CleanButton.protoType = {
    setCategoties: PropTypes.func.isRequired
}
