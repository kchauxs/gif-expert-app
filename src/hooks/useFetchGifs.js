import { useEffect, useState } from 'react'
import { getGifs } from '../helpers/getGisf'

export const useFetchGif = (category) => {

    const [state, setstate] = useState({
        data: [],
        loading: true
    })

    useEffect(() => {
        getGifs(category)
            .then(img => {
                setTimeout(() => {
                    setstate({
                        data: img,
                        loading: false
                    })
                }, 500);
            })
    }, [category])

    return state

}