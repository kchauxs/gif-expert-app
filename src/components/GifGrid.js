import React, { Fragment } from 'react'
import { useFetchGif } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'
import logo from '../logo.svg';
// import { getGifs } from '../helpers/getGisf'

export const GifGrid = ({ category }) => {
    
    // const [images, setImages] = useState([])

    // useEffect(() => {
    //     getGifs(category)
    //         .then(setImages)
    // }, [category])


    const { data:images, loading } = useFetchGif(category)


    return (
        <Fragment>
            <h3 className="animate__bounceIn">{category}</h3>

            {/* {loading && <p>Cargando...</p>} */}
            {loading &&  <img src={logo} className="App-logo" alt="logo" />}

            <div className="card-grid">

                {
                    images.map(img => <GifGridItem key={img.id} {...img} />)
                }
            </div>
        </Fragment>
    )
}
