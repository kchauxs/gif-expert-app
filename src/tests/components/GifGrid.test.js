import React from 'react'
import "@testing-library/jest-dom"
import { shallow } from "enzyme"
import { GifGrid } from '../../components/GifGrid'
import { useFetchGif } from '../../hooks/useFetchGifs'

jest.mock('../../hooks/useFetchGifs')

describe('<GifGrid/>', () => {

    const categoria = "boo"


    test('should: mostrar correctamente el componente GifGrid', () => {

        useFetchGif.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow(<GifGrid category={categoria} />)
        expect(wrapper).toMatchSnapshot()
    })

    test('should: mostar el texto de la categoria ', () => {
        const wrapper = shallow(<GifGrid category={categoria} />)
        const h3 = wrapper.find("h3")
        expect(h3.text().trim()).toBe(categoria)
    })


    test('should: mostar los items cuando se cargan las imagenes', () => {

        const gifs = [{
            id:"AA111",
            url:"http://localhost/test.jpg",
            title:"cualquier cosa"
        },
        {
            id:"AA122",
            url:"http://localhost/test.jpg",
            title:"cualquier cosa"
        }]

        useFetchGif.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category={categoria} />)
        expect(wrapper).toMatchSnapshot() 
        expect(wrapper.find("img").exists()).toBe(false)
        expect(wrapper.find("GifGridItem").length).toBe(gifs.length)
    }) 



})
