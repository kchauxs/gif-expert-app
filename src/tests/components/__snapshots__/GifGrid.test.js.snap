// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid/> should: mostar los items cuando se cargan las imagenes 1`] = `
<Fragment>
  <h3
    className="animate__bounceIn"
  >
    boo
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="AA111"
      key="AA111"
      title="cualquier cosa"
      url="http://localhost/test.jpg"
    />
    <GifGridItem
      id="AA122"
      key="AA122"
      title="cualquier cosa"
      url="http://localhost/test.jpg"
    />
  </div>
</Fragment>
`;

exports[`<GifGrid/> should: mostrar correctamente el componente GifGrid 1`] = `
<Fragment>
  <h3
    className="animate__bounceIn"
  >
    boo
  </h3>
  <img
    alt="logo"
    className="App-logo"
    src="logo.svg"
  />
  <div
    className="card-grid"
  />
</Fragment>
`;
